src/groovy/org/pih/warehouse/api/StockMovementItem.groovy

```diff
@@ -199,15 +199,14 @@ class StockMovementItem {
 
 
     static StockMovementItem createFromTokens(String [] tokens) {
-        String requisitionItemId = tokens[0] ?: null
-        String productCode = tokens[1] ?: null
-        String productName = tokens[2] ?: null
-        String palletName = tokens[3] ?: null
-        String boxName = tokens[4] ?: null
-        String lotNumber = tokens[5] ?: null
-        Date expirationDate = tokens[6] ? Constants.EXPIRATION_DATE_FORMATTER.parse(tokens[6]) : null
-        Integer quantityRequested = tokens[7] ? tokens[7].toInteger() : null
-        String recipientId = tokens[8]
+        String productCode = tokens[0] ?: null
+        String productName = tokens[1] ?: null
+        String palletName = tokens[2] ?: null
+        String boxName = tokens[3] ?: null
+        String lotNumber = tokens[4] ?: null
+        Date expirationDate = tokens[5] ? Constants.EXPIRATION_DATE_FORMATTER.parse(tokens[6]) : null
```

Comment:
This would cause a bug because you check if tokens[5] is not null then try to convert tokens[6] to a date. This would throw an exception because tokens[6] is an integer.